### PR TITLE
Log drains: delete command

### LINKED ISSUE #554: Remove a drain
Remove a drain by application

```bash
scalingo --app my-app logs-drains-remove <id>
```

### PR DESCRIPTION
**Pull request by curzolapierre**
Add remove command

Example:
```bash
$ scalingo log-drains-remove --app sample-go-martini tcp+tls://logs.papertrailapp.com:12345
-----> The log drain: tcp+tls://logs.papertrailapp.com:12345 has been deleted
```

Fix #554

### CODE REVIEW

**Comment by Soulou** [comment]
Why does the user need to use an URL? Didn't we agree, it should be "easier", using flags etc.?

**Comment by curzolapierre** [comment]
@Soulou We planned to use uuid for the MVP. But as we are only on the first step of the MVP here, without uuid implemented yet, we (with @EtienneM) decided to use urls returned by the list command.
I agree on the point it is less consistent to use url, but flags are usefull to control url generation and are less relevant in case of deletion.
What do you think ?

**Comment by Soulou** [comment]
So if I understand this well, the problem is that we don't have a unique identifier for each log drain, so we can't ask users to use them for the deletion, right?

**Comment by curzolapierre** [comment]
I hadn't thought about that, the logs service will allow to add multiple same url by application ?
The current implementation of addition on the API will return an error if the user try to add an existing url.

File: cmd/autocomplete/log_drains_remove.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		return nil
+	}
+
+	client, err := config.ScalingoClient()
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
+	}
+	drains, err := client.LogDrainsList(appName)
+	if err == nil {
+
```
Comment (EtienneM) [inline cmd/autocomplete/log_drains_remove.go:23]:
```suggestion
```

File: cmd/autocomplete/log_drains_remove.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	if appName == "" {
+		return nil
+	}
+
+	client, err := config.ScalingoClient()
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
+	}
+	drains, err := client.LogDrainsList(appName)
+	if err == nil {
```
Comment (EtienneM) [inline cmd/autocomplete/log_drains_remove.go:22]: We usually compare `err` with `!=nil`. It's odd to do the other way around. Can you modify this piece of code to compare `if err != nil` first? And add a comment to explain why you ignore the error please

File: cmd/log_drains.go [hunk 3]
```diff
@@ -80,4 +80,34 @@ var (
...
+	logDrainsRemoveCommand = cli.Command{
+		Name:     "log-drains-remove",
+		Category: "Log drains",
+		Flags:    []cli.Flag{appFlag},
+		Usage:    "Remove a log drain from an application",
+		Description: `Remove a log drain from an application:
+
+	$ scalingo --app my-app log-drains-remove syslog://custom.logstash.com:12345
+
+	# See also commands 'log-drains-add', 'log-drains' and 'log-drains'`,
```
Comment (EtienneM) [inline cmd/log_drains.go:93]:
```suggestion
	# See also commands 'log-drains-add', 'log-drains'`,
```

File: log_drains/remove.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+func Remove(app string, URL string) error {
+	c, err := config.ScalingoClient()
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
+	}
+
+	err = c.LogDrainRemove(app, URL)
+	if err != nil {
+		return errgo.Mask(err)
```
Comment (EtienneM) [inline log_drains/remove.go:17]:
```suggestion
		return errgo.Notef(err, "fail to remove the log drain from the application")
```

File: log_drains/remove.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"github.com/Scalingo/cli/config"
+	"github.com/Scalingo/cli/io"
+	"gopkg.in/errgo.v1"
+)
+
+func Remove(app string, URL string) error {
+	c, err := config.ScalingoClient()
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
```
Comment (EtienneM) [inline log_drains/remove.go:12]:
```suggestion
		return errgo.Notef(err, "fail to get Scalingo client to remove a log drain from the application")
```

File: cmd/autocomplete/log_drains_remove.go (hunk 2 above)
Comment (curzolapierre) [reply cmd/autocomplete/log_drains_remove.go:22]: Silly copy past from another remove autocomplete my bad. I'll fix the other too

File: Gopkg.toml [hunk 6]
```diff
@@ -26,7 +26,7 @@
 
 [[constraint]]
   name = "github.com/Scalingo/go-scalingo"
-  version = "^4"
+  branch = "feature/cli/554/add_log_drains_del_cmd"
```
Comment (EtienneM) [inline Gopkg.toml:29]:
```suggestion
  version = "^4"
```

File: Gopkg.lock [hunk 7]
```diff
@@ -37,8 +38,7 @@
     "io",
   ]
   pruneopts = "NUT"
-  revision = "9f30cefb0ea118d42f84c30964bbb3a842f004be"
-  version = "v4.5.2"
```
Comment (EtienneM) [inline Gopkg.lock:41]: I think you didn't run `dep ensure -update github.com/Scalingo/go-scalingo` as this line should be updated to `v4.5.3`

**Pull request merged**